taxi/forms.py

```diff
@@ -3,50 +3,79 @@
 from django.contrib.auth.forms import UserCreationForm
 from django.core.exceptions import ValidationError
 
-from taxi.models import Car, Driver
+from taxi.models import Driver, Car
+
+
+class DriverCreationForm(UserCreationForm):
+
+    class Meta(UserCreationForm.Meta):
+        model = Driver
+        fields = UserCreationForm.Meta.fields + (
+            "first_name", "last_name", "license_number",)
+
+
+class DriverLicenseUpdateForm(forms.ModelForm):
+    LENGTH = 8
+
+    class Meta(UserCreationForm):
+        model = Driver
+        fields = ("license_number",)
+
+    def clean_license_number(self):
+        license_number = self.cleaned_data["license_number"]
+
+        if len(license_number) != DriverLicenseUpdateForm.LENGTH:
+            raise ValidationError(
+                f"Ensure that license number length is "
+                f"{DriverLicenseUpdateForm.LENGTH}"
+            )
+        if not license_number[:3].isupper() or (
+                not license_number[:3].isalpha()):
+            raise ValidationError(
+                "Ensure that first 3 letters are uppercase letters"
+            )
+
+        if not license_number[-5:].isdigit():
+            raise ValidationError(
+                "Ensure that last 5 characters are digits"
+            )
+        return license_number
 
 
 class CarForm(forms.ModelForm):
     drivers = forms.ModelMultipleChoiceField(
         queryset=get_user_model().objects.all(),
         widget=forms.CheckboxSelectMultiple,
+        required=False,
     )
 
     class Meta:
         model = Car
         fields = "__all__"
 
 
-class DriverCreationForm(UserCreationForm):
-    class Meta(UserCreationForm.Meta):
-        model = Driver
-        fields = UserCreationForm.Meta.fields + (
-            "license_number",
-            "first_name",
-            "last_name",
-        )
-
-    def clean_license_number(self):  # this logic is optional, but possible
-        return validate_license_number(self.cleaned_data["license_number"])
-
-
-class DriverLicenseUpdateForm(forms.ModelForm):
-    class Meta:
-        model = Driver
-        fields = ["license_number"]
+class CarSearchForm(forms.Form):
+    model = forms.CharField(
+        max_length=255,
+        required=False,
+        label="",
+        widget=forms.TextInput(attrs={"placeholder": "Search by model.."})
```

Comment:
why do you use 2 dots? I think that 1 or 3 would be better. It is not a mistake, but it can be changed 😅